File: LeetCode/cpp/1886_Determine_Whether_Matrix_Can_Be_Obtained_By_Rotation.cpp
class Solution {
public:
	bool check(vector<vector<int>>& target, vector<vector<int>> sample) {
		for (int i = 0; i < target.size(); i++) {
			for (int j = 0; j < target[i].size(); j++) {
				if (target[i][j] != sample[i][j]) {
					return false;
				}
			}
		}

		return true;
	}

	bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
		vector<vector<int>> v1(mat.size(), vector<int>(mat.size()));
		vector<vector<int>> v2(mat.size(), vector<int>(mat.size()));
		vector<vector<int>> v3(mat.size(), vector<int>(mat.size()));

		vector<int> temp;
		for (int i = 0; i < mat.size(); i++) {
			temp.clear();
			for (int j = 0; j < mat[i].size(); j++) {
				temp.push_back(mat[i][j]);
			}

			int temp_idx = 0;
			int end_point = mat[i].size() - 1 - i;
			for (int j = 0; j < mat[i].size(); j++) {
				v1[j][end_point] = temp[temp_idx];
				temp_idx++;
			}

			temp_idx = 0;
			end_point = mat[i].size() - 1 - i;
			for (int j = mat[i].size() - 1; j >= 0; j--) {
				v2[end_point][j] = temp[temp_idx];
				temp_idx++;
			}

			temp_idx = 0;
			end_point = i;
			for (int j = mat[i].size() - 1; j >= 0; j--) {
				v3[j][end_point] = temp[temp_idx];
				temp_idx++;
			}
		}

		if (check(target, mat) || check(target, v1) || check(target, v2) || check(target, v3)) {
			return true;
		}

		return false;
	}
};

File: LeetCode/java/1886_Determine_Whether_Matrix_Can_Be_Obtained_By_Rotation.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    private boolean check(int[][] target, int[][] sample) {
        for (int i = 0; i < target.length; i++) {
            for (int j = 0; j < target[i].length; j++) {
                if (target[i][j] != sample[i][j]) {
                    return false;
                }
            }
        }

        return true;
    }

    public boolean findRotation(int[][] mat, int[][] target) {
        int[][] v1 = new int[mat.length][mat.length];
        int[][] v2 = new int[mat.length][mat.length];
        int[][] v3 = new int[mat.length][mat.length];

        List<Integer> temp = new ArrayList<>();
        for (int i = 0; i < mat.length; i++) {
            temp.clear();
            for (int j = 0; j < mat[i].length; j++) {
                temp.add(mat[i][j]);
            }

            int tempIdx = 0;
            int endPoint = mat[i].length - 1 - i;
            for (int j = 0; j < mat[i].length; j++) {
                v1[j][endPoint] = temp.get(tempIdx);
                tempIdx++;
            }

            tempIdx = 0;
            endPoint = mat[i].length - 1 - i;
            for (int j = mat[i].length - 1; j >= 0; j--) {
                v2[endPoint][j] = temp.get(tempIdx);
                tempIdx++;
            }

            tempIdx = 0;
            endPoint = i;
            for (int j = mat[i].length - 1; j >= 0; j--) {
                v3[j][endPoint] = temp.get(tempIdx);
                tempIdx++;
            }
        }

        if (check(target, mat) || check(target, v1) || check(target, v2) || check(target, v3)) {
            return true;
        }

        return false;
    }
}